CONTRIBUTING.md

```diff
@@ -0,0 +1,257 @@
+   
+### Integration tests
+
+    mvn clean verify
+
+### Integration tests native
+
+    mvn clean verify -Dnative
+    
+### Generating documentation
+   
+    mvn clean package -Prelease    
+    
+
+## Build
+
+* Clone the repository: `git clone git@github.com:datastax/cassandra-quarkus.git`
+* Navigate to the directory: `cd cassandra-quarkus`
+* Invoke `mvn clean install -DskipTests=true -Dmaven.javadoc.skip=true -B -V` from the root directory
+
+```bash
+git clone git@github.com:datastax/cassandra-quarkus.git
+cd cassandra-quarksu
+mvn clean install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
+```
+
+This build skipped all the tests, native-image builds and documentation generation. 
+
+Removing the `-DskipTests -DskipITs` flags enables the tests. 
+It will take much longer to build but will give you more guarantees on your code. 
+
+You can build and test native images in the integration tests supporting it by using `mvn install -Dnative`.
+
+
+## License headers
+
+The build will fail if some license headers are missing. To update all files from the command line,
+run:
+
+```
+mvn license:format
+```
+## Commits
+
+Keep your changes **focused**. Each commit should have a single, clear purpose expressed in its 
+message.
+
+Resist the urge to "fix" cosmetic issues (add/remove blank lines, move methods, etc.) in existing
+code. This adds cognitive load for reviewers, who have to figure out which changes are relevant to
+the actual issue. If you see legitimate issues, like typos, address them in a separate commit (it's
+fine to group multiple typo fixes in a single commit).
+
+Isolate trivial refactorings into separate commits. For example, a method rename that affects dozens
+of call sites can be reviewed in a few seconds, but if it's part of a larger diff it gets mixed up
+with more complex changes (that might affect the same lines), and reviewers have to check every
+line.
+
+Commit message subjects start with a capital letter, use the imperative form and do **not** end
+with a period:
+
+* correct: "Add test for CQL request handler"
+* incorrect: "~~Added test for CQL request handler~~"
+* incorrect: "~~New test for CQL request handler~~"
+
+Avoid catch-all messages like "Minor cleanup", "Various fixes", etc. They don't provide any useful
+information to reviewers, and might be a sign that your commit contains unrelated changes.
+ 
+We don't enforce a particular subject line length limit, but try to keep it short.
+
+You can add more details after the subject line, separated by a blank line. The following pattern
+(inspired by [Netty](http://netty.io/wiki/writing-a-commit-message.html)) is not mandatory, but
+welcome for complex changes:
+
+```
+One line description of your change
+ 
+Motivation:
+
+Explain here the context, and why you're making that change.
+What is the problem you're trying to solve.
+ 
+Modifications:
+
+Describe the modifications you've done.
+ 
+Result:
+
+After your change, what will change.
+```
+
```

Comment:
```suggestion

### Referencing issues in commits

If your pull request fixes an open issue, please include its number at the beginning of the commit message, e.g.

    #62: Enforce package name conventions 

```

Reply:
The format I'm proposing above for commit titles is open for debate.

Reply:
Creating a new branch name with leading `#`  is problematic.
If you do:
```
git checkout -b #60
```
you will get an error:
```
error: switch `b' requires a value
```

Reply:
I was referring to commit messages, not branch names. GitHub requires you to introduce the issue number by a `#`.

For branch names, I think `issue62` is good enough, wdyt?